tests: adc: imx93_evk/mimx9352/m33 : build failure

### Describe the bug

- imx93_evk/mimx9352/m33

commit e7d24c80884fc4fda99c378b7b57551196de73e5
Author: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>
Date:   Fri Dec 26 13:28:16 2025 +0800

    drivers: adc: add clock frequency selection for SAR ADC
    
    This commit enhances the SAR ADC driver by adding
    support for clock frequency selection. The clock
    frequency can now be configured via the devicetree,
    allowing for better optimization of ADC performance
    based on application requirements.
    
    Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

 drivers/adc/adc_nxp_sar_adc.c     | 11 ++++++++---
 dts/bindings/adc/nxp,sar-adc.yaml | 12 ++++++++++++
 2 files changed, 20 insertions(+), 3 deletions(-)

### Regression

- [x] This is a regression.

### Steps to reproduce

west build -b imx93_evk/mimx9352/m33 tests/drivers/adc/adc_api -T drivers.adc.b_u585i_iot02a_adc4

### Relevant log output

```shell
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/adc/adc_nxp_sar_adc.c:468:23: error: 'ADC_MCR_ADCLKSEL_MASK' undeclared (first use in this function); did you mean 'ADC_MCR_ADCLKSE_MASK'?
  468 |                       ADC_MCR_ADCLKSEL_MASK)) | ADC_MCR_OWREN(config->overwrite ? 1U : 0U) |
      |                       ^~~~~~~~~~~~~~~~~~~~~
      |                       ADC_MCR_ADCLKSE_MASK
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/adc/adc_nxp_sar_adc.c:468:23: note: each undeclared identifier is reported only once for each function it appears in
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/adc/adc_nxp_sar_adc.c:470:23: error: implicit declaration of function 'ADC_MCR_ADCLKSEL'; did you mean 'ADC_MCR_ADCLKSE'? [-Werror=implicit-function-declaration]
  470 |                       ADC_MCR_ADCLKSEL(config->conv_clk_freq_div_factor));
      |                       ^~~~~~~~~~~~~~~~
      |                       ADC_MCR_ADCLKSE
cc1: all warnings being treated as errors
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: ubuntu
- toolchain: zephyr
- version: v4.3.0-3362-g1428bc537a39

### Additional Context

_No response_

--- a/drivers/adc/adc_nxp_sar_adc.c
+++ b/drivers/adc/adc_nxp_sar_adc.c
@@ -31,6 +31,17 @@ LOG_MODULE_REGISTER(nxp_sar_adc, CONFIG_ADC_LOG_LEVEL);
 #define NXP_SAR_ADC_HAS_GROUP2_REGS 0
 #endif
 
+/* Some NXP SAR ADC variants don't implement MCR[ADCLKSEL].
+ * Guard both the mask and the value macro so the driver can compile across SoCs.
+ */
+#if defined(ADC_MCR_ADCLKSEL_MASK) && defined(ADC_MCR_ADCLKSEL)
+#define NXP_SAR_ADC_MCR_ADCLKSEL_MASK ADC_MCR_ADCLKSEL_MASK
+#define NXP_SAR_ADC_MCR_ADCLKSEL(x)   ADC_MCR_ADCLKSEL(x)
+#else
+#define NXP_SAR_ADC_MCR_ADCLKSEL_MASK 0U
+#define NXP_SAR_ADC_MCR_ADCLKSEL(x)   0U
+#endif
+
 struct nxp_sar_adc_config {
 	ADC_Type *base;
 	const struct device *clock_dev;
@@ -465,9 +476,9 @@ static int nxp_sar_adc_init(const struct device *dev)
 	}
 
 	base->MCR = ((base->MCR & ~(ADC_MCR_OWREN_MASK | ADC_MCR_ACKO_MASK |
-		      ADC_MCR_ADCLKSEL_MASK)) | ADC_MCR_OWREN(config->overwrite ? 1U : 0U) |
+		      NXP_SAR_ADC_MCR_ADCLKSEL_MASK)) | ADC_MCR_OWREN(config->overwrite ? 1U : 0U) |
 		      ADC_MCR_ACKO(config->auto_clock_off ? 1U : 0U) |
-		      ADC_MCR_ADCLKSEL(config->conv_clk_freq_div_factor));
+		      NXP_SAR_ADC_MCR_ADCLKSEL(config->conv_clk_freq_div_factor));
 
 	/* Disable global and all channels' interrupt. */
 	base->IMR = 0U;

